dev/com.ibm.ws.session/src/com/ibm/ws/session/http/AbstractHttpSession.java

```diff
@@ -0,0 +1,239 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.session.http;
+
+import java.util.Enumeration;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpSession;
+import javax.servlet.http.HttpSessionBindingListener;
+
+import com.ibm.ws.session.utils.LoggingUtil;
+import com.ibm.wsspi.session.ISession;
+
+/**
+ * This class provides the adapted version of the ISession.
+ * It simply wrappers the session and proxies any of its method calls to
+ * the underlying ISession object.
+ * <p> 
+ * Since Servlet 6.0 : 
+ *      These are common APIs moved up from HttpSessionImpl.
+ *      Additional APIs for a specific servlet will be in subclasses
+ *  
+ */
+public abstract class AbstractHttpSession implements HttpSession {
+
+    /*
+     * A reference to the wrapper iManagedSession object
+     */
+    protected ISession _iSession;
+
+    /*
+     * A reference to the ServletContext object returns for all http sessions
+     * created in this SessionManager.
+     */
+    protected ServletContext _servletContext;
+
+    /*
+     * Adding a message for when IllegalStateException is thrown
+     */
+    private static final String iseMessage = "The method is called on an invalidated session: ";
+
```

Comment:
Again I would suggest an AbstractHttpSession constructor that the subclasses can delegate to.

Reply:
added constructor for abstract classes and called super() from its derived classes